src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -112,4 +116,72 @@ public function handleChargeSucceeded(Event $event): Response
 
         return $this->respondWithData($event->data->object);
     }
+
+    public function handlePayoutPaid(Event $event)
+    {
+        $stripeClient = new \Stripe\StripeClient($this->apiKey);
+
+        $payoutId = $event->data->object->id;
+
+        $this->logger->info('Getting all charges related to payout Id: ' . $payoutId);
+
+        $hasMore = true;
+        $lastBalanceTransactionId = null;
+        $paidChargeIds = [];
+        $attributes = [
+            'limit' => 100,
+            'payout' => $payoutId,
+            'type' => 'charge',
+        ];
+
+        while ($hasMore) {
+            $balanceTransactions = $stripeClient->balanceTransactions->all($attributes);
+
+            foreach ($balanceTransactions->data as $balanceTransaction) {
+                $paidChargeIds[] = $balanceTransaction->source;
+                $lastBalanceTransactionId = $balanceTransaction->id;
+            }
+
+            $hasMore = $balanceTransactions->has_more;
+
+            // We get a Stripe exception if we start this with a null or empty value,
+            // so we only include this if there's more items to iterate.
+            if ($hasMore) {
+                $attributes['start_after'] = $lastBalanceTransactionId;
+            }
+        }
+        $this->logger->info('Getting all paid charge Ids complete, found: ' . sizeof($paidChargeIds));
+
+        if (sizeof($paidChargeIds) > 0) {
+            $count = 0;
+
+            foreach ($paidChargeIds as $Id) {
+                /** @var Donation $donation */
+                $donation = $this->donationRepository->findOneBy(['chargeId' => $Id]);
+
+                if ($donation) {
+                    if ($donation->getDonationStatus() === 'Collected') {
+                        // We're confident to set donation status to paid because this
+                        // method is called only when Stripe event `payout.paid` is received.
+                        $donation->setDonationStatus('Paid');
+
+                        $this->entityManager->persist($donation);
+                        $this->donationRepository->push($donation, false);
+
+                        $count++;
+                    } else {
+                        $this->logger->error('Unexpected donation status found for charge Id: ' . $Id);
```

Comment:
Surely donations already _Paid_ should not error? Remember that webhooks can be re-sent and delivery order is not guaranteed.